Sources/Commands/Options.swift

```diff
@@ -246,13 +246,28 @@ public struct SwiftToolOptions: ParsableArguments {
     @Flag(name: [.long, .customLong("disable-automatic-resolution")], help: "Disable automatic resolution if Package.resolved file is out-of-date")
     var forceResolvedVersions: Bool = false
 
-    @Flag(name: .customLong("index-store"), inversion: .prefixedEnableDisable, help: "Enable or disable  indexing-while-building feature")
-    var indexStoreEnable: Bool?
-        
+    // @Flag works best when there is a default value present
+    // if true, false aren't enough and a third state is needed
+    // nil should not be the goto. Instead create an enum
+    enum StoreMode: String, EnumerableFlag {
+        case autoIndexStore
+        case enableIndexStore
+        case disableIndexStore
+    }
+    
+    @Flag(help: "Enable or disable indexing-while-building feature")
+    var indexStoreMode: StoreMode = .autoIndexStore
+    
     /// The mode to use for indexing-while-building feature.
     var indexStore: BuildParameters.IndexStoreMode {
```

Comment:
this feels like it can be an extension on `StoreMode` instead of a top level thing

Reply:
But an enum can't have stored properties?

Reply:
> But an enum can't have stored properties?

but, maybe associated-values will help... https://www.hackingwithswift.com/sixty/2/9/enum-associated-values

Reply:
just a function would do